L1Trigger/L1THGCal/plugins/ntuples/HGCalTriggerNtupleHGCDigis.cc

```diff
@@ -173,11 +218,33 @@ fill(const edm::Event& e, const edm::EventSetup& es)
           hgcdigi_simenergy_.emplace_back(hit_energy); 
         }
       }
+
+     for(const auto& digi : bh_digis)
+      {
+        const HcalDetId id(digi.id());
+        bhdigi_id_.emplace_back(id.rawId());
+        bhdigi_subdet_.emplace_back(id.subdetId());
+        bhdigi_side_.emplace_back(id.zside());
+        bhdigi_layer_.emplace_back(id.depth());
+        bhdigi_ieta_.emplace_back(id.ieta());
+        bhdigi_iphi_.emplace_back(id.iphi());
+        GlobalPoint cellpos = triggerGeometry_->bhGeometry().getPosition(id.rawId());
+        bhdigi_eta_.emplace_back(cellpos.eta());
+        bhdigi_phi_.emplace_back(cellpos.phi());
+        bhdigi_z_.emplace_back(cellpos.z());
+        bhdigi_data_.emplace_back(digi[2].data()); 
```

Comment:
is `2` the sample of interest in the digi? I guess SOI isn't a parameter of the (rather bare-bones) `HGCDataFrame` class; something to consider for the future in lieu of hardcoding the value everywhere.